src/file/tokio_compat_file.rs

```diff
@@ -469,6 +474,12 @@ impl AsyncWrite for TokioCompatFile {
             .map(|n| min(n, max_write_len))
             .unwrap_or(max_write_len);
```

Comment:
Idea: We can write part of the buffer to fill up the buffer as much as possible to improve efficiency: